service/kernel/error_handling.cpp

```diff
@@ -105,19 +105,21 @@ class MessageCollection : public Collection<SharedPtr<Message<IDType> > >
         bool found = false;
         size_t index = 0;
 
-        if(this->size() == 0) { return find(_noMessageFound); }
+        if (this->size() == 0)
+        {
+            return SharedPtr<Message<IDType>>(new Message<IDType>(_noMessageFound, "No error message found"));
```

Comment:
NoErrorMessageFound should be used here.

Reply:
Fixed